pkg/cmd/workflow/view/http.go

```diff
@@ -0,0 +1,58 @@
+package view
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	runShared "github.com/cli/cli/pkg/cmd/run/shared"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+)
+
+type workflowRuns struct {
+	Total int
+	Runs  []runShared.Run
+}
+
+func getWorkflowContent(client *api.Client, repo ghrepo.Interface, ref string, workflow *shared.Workflow) (string, error) {
+	path := fmt.Sprintf("repos/%s/contents/%s", ghrepo.FullName(repo), workflow.Path)
+	if ref != "" {
+		q := fmt.Sprintf("?ref=%s", url.QueryEscape(ref))
+		path = path + q
+	}
+
+	type Result struct {
+		Content string
+	}
+
+	var result Result
+	err := client.REST(repo.RepoHost(), "GET", path, nil, &result)
+	if err != nil {
+		return "", err
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(result.Content)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode workflow file: %w", err)
+	}
+
+	return string(decoded), nil
+}
+
+func getWorkflowRuns(client *api.Client, repo ghrepo.Interface, workflow *shared.Workflow) (workflowRuns, error) {
```

Comment:
Initially I was using `shared.GetRunsByWorkflow()` since they are mostly doing the same thing but I also wanted to get the total run count which is not returned by that function. Since `shared.GetRunsByWorkflow()` is used in a couple different spots I thought it best not to change its return value at this time.